packages/datadog-plugin-sharedb/src/index.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const MessagesAwaitingResponse = new WeakMap();
+const READABLE_ACTION_NAMES = {
+  hs: 'handshake',
+  qf: 'query-fetch',
+  qs: 'query-subscribe',
+  qu: 'query-unsubscribe',
+  bf: 'bulk-fetch',
+  bs: 'bulk-subscribe',
+  bu: 'bulk-unsubscribe',
+  f: 'fetch',
+  s: 'subscribe',
+  u: 'unsubscribe',
+  op: 'op',
+  nf: 'snapshot-fetch',
+  nt: 'snapshot-fetch-by-ts',
+  p: 'presence-broadcast',
+  pr: 'presence-request',
+  ps: 'presence-subscribe',
+  pu: 'presence-unsubscribe'
+};
+
+function createTraceName(action, collection) {
+  let actionName = READABLE_ACTION_NAMES[action];
```

Comment:
Since operations are composed of multiple actions, I don't think there is a one to one mapping between spans and actions. I would expect for example for a `sharedb.request` to wrap the entire lifetime of the request between the `receive` and `reply` actions, and something similar like `sharedb.query` for queries and any other types of communication supported by ShareDB.

Reply:
I think you're mixing up the concept of `ops` and `action`.

My understanding is the server receives an event with a single action, which may result in many ops.

What you are saying you expect is in fact what this does! :) The span wrap the entire lifetime of receiving a message, what ever happens in between (creating ops, subscriptions to redis, etc), and sending the response.

`query` will simply send a message, which also goes through this mechanism.

Reply:
> query will simply send a message, which also goes through this mechanism.
Any way to identify queries specifically? I feel like `sharedb.request` somewhat hides the fact that it's a query, but let me know if it actually makes sense to group everything together from a ShareDB user perspective.